pkg/keda/scaleobject_test.go

```diff
@@ -0,0 +1,593 @@
+			assert.Equal(t, testCase.expectedError, err.Error())
+		} else {
+			assert.Equal(t, testCase.testScaleTargetRef.Name,
+				testCase.testScaleObject.Definition.Spec.ScaleTargetRef.Name)
+		}
+	}
+}
+
+func TestScaleObjectWithScaleTargetRef(t *testing.T) {
+	testCases := []struct {
+		testScaleTargetRef kedav2v1alpha1.ScaleTarget
+		expectedError      bool
+		expectedErrorText  string
+	}{
+		{
+			testScaleTargetRef: kedav2v1alpha1.ScaleTarget{
+				Name: "test-app",
+			},
+			expectedError:     false,
+			expectedErrorText: "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidScaledObjectBuilder(buildScaledObjectClientWithDummyObject())
+
+		result := testBuilder.WithScaleTargetRef(testCase.testScaleTargetRef)
+
+		if testCase.expectedError {
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, result.errorMsg)
+			}
+		} else {
+			assert.NotNil(t, result)
+			assert.Equal(t, testCase.testScaleTargetRef, *result.Definition.Spec.ScaleTargetRef)
+		}
+	}
+}
+
+func TestScaleObjectWithPollingInterval(t *testing.T) {
+	testCases := []struct {
+		testPollingInterval int32
+		expectedError       bool
+		expectedErrorText   string
+	}{
+		{
+			testPollingInterval: pollingInterval,
+			expectedError:       false,
+			expectedErrorText:   "",
+		},
+		{
+			testPollingInterval: zeroValue,
+			expectedError:       false,
+			expectedErrorText:   "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidScaledObjectBuilder(buildScaledObjectClientWithDummyObject())
+
+		result := testBuilder.WithPollingInterval(testCase.testPollingInterval)
+
+		if testCase.expectedError {
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, result.errorMsg)
+			}
+		} else {
+			assert.NotNil(t, result)
+			assert.Equal(t, testCase.testPollingInterval, *result.Definition.Spec.PollingInterval)
+		}
+	}
+}
+
+func TestScaleObjectWithCooldownPeriod(t *testing.T) {
+	testCases := []struct {
+		testCooldownPeriod int32
+		expectedError      bool
+		expectedErrorText  string
+	}{
+		{
+			testCooldownPeriod: cooldownPeriod,
+			expectedError:      false,
+			expectedErrorText:  "",
+		},
+		{
+			testCooldownPeriod: zeroValue,
+			expectedError:      false,
+			expectedErrorText:  "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidScaledObjectBuilder(buildScaledObjectClientWithDummyObject())
+
+		result := testBuilder.WithCooldownPeriod(testCase.testCooldownPeriod)
+
+		if testCase.expectedError {
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, result.errorMsg)
+			}
+		} else {
+			assert.NotNil(t, result)
+			assert.Equal(t, testCase.testCooldownPeriod, *result.Definition.Spec.CooldownPeriod)
+		}
+	}
+}
+
+func TestScaleObjectWithMinReplicaCount(t *testing.T) {
+	testCases := []struct {
+		testMinReplicaCount int32
+		expectedError       bool
```

Comment:
Do you plan to add more testcases ? Otherwise only one of the 2 `expectedError` / `expectedErrorText` could be used